src/session/session_api.c

```diff
@@ -649,7 +649,9 @@ __wt_open_cursor(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, co
     hash_value = 0;
 
     /* We should not open other cursors when there are open history store cursors in the session. */
-    WT_ASSERT(session, strcmp(uri, WT_HS_URI) == 0 || session->hs_cursor_counter == 0);
+    WT_ASSERT(session,
```

Comment:
This was updated to make sure another HS cursor is only opened when the flag `WT_BTREE_VERIFY` is set.
@agorrod suggested another solution which would be to reuse the HS cursor of the session if it exists.

Reply:
There is one more issue here: when the call comes from `conn_api.c:wiredtiger_open`, the assertion still fires.
We could rely on the session name that is set to `verify hs` but it does not sound robust. Since we are using an internal session for this call, we could check this as well.